x-pack/plugins/task_manager/server/monitoring/workload_statistics.ts

```diff
@@ -273,16 +273,22 @@ export function createWorkloadAggregator({
       let totalCost = 0;
       const taskTypeSummary = taskTypes.reduce((acc, bucket) => {
         const value = bucket as TaskTypeWithStatusBucket;
-        const cost =
-          value.doc_count * taskDefinitions.get(value.key as string)?.cost ?? TaskCost.Normal;
-        totalCost += cost;
-        return Object.assign(acc, {
-          [value.key as string]: {
-            count: value.doc_count,
-            cost,
-            status: mapValues(keyBy(value.status.buckets, 'key'), 'doc_count'),
-          },
-        });
+        try {
+          const taskDef = taskDefinitions.get(value.key as string);
+          const cost = value.doc_count * taskDef?.cost ?? TaskCost.Normal;
+
+          totalCost += cost;
+          return Object.assign(acc, {
+            [value.key as string]: {
+              count: value.doc_count,
+              cost,
+              status: mapValues(keyBy(value.status.buckets, 'key'), 'doc_count'),
+            },
+          });
+        } catch (err) {
+          // task type is not registered with dictionary, do not add to summary
+          return acc;
```

Comment:
I don't know if we want to keep the unrecognized task type in the health summary? This change would remove it altogether whereas previously the health report would show it, even if task manager would never claim it.